src/app/body/dashboard/testimonials-dashboard/testimonials-dashboard.component.ts

```diff
@@ -80,4 +81,36 @@ submit(){
   }
 }
 
+editTestimonial(Uid: any, Achievement: any , ImageUrl:any , Name:any , Testimonial:any){
+  if(this.uploadService.testimonialUrl){
+    this.testimonial.ImageUrl=this.uploadService.testimonialUrl;
+     }
+     else{
+       alert("No file Uploaded")
+     }
+  const callable = this.functions.httpsCallable('testimonials/editTestimonial');
+    callable({Uid:Uid, Achievement: Achievement, ImageUrl: this.testimonial.ImageUrl, Name: Name, Testimonial: Testimonial}).subscribe({
+      next: (data) => {
+        console.log("testimonial edited successfully");
```

Comment:
Implement a pop up saying edit successful or delete successful after editing or deleting and refresh the data

Reply:
Done !